sdk/python/foundation-models/system/finetune/question-answering/extractive-qa.ipynb

```diff
@@ -344,6 +344,56 @@
     "Create the job that uses the `question-answering` pipeline component. [Learn more](https://github.com/Azure/azureml-assets/blob/main/training/finetune_acft_hf_nlp/components/pipeline_components/question_answering/README.md) about all the parameters supported for fine tuning."
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Define finetune parameters\n",
+    "\n",
+    "Finetune parameters can be grouped into 2 categories - training parameters, optimization parameters\n",
+    "\n",
+    "Training parameters define the training aspects such as - \n",
+    "1. the optimizer, scheduler to use\n",
+    "2. the metric to optimize the finetune\n",
+    "3. number of training steps and the batch size\n",
+    "and so on\n",
+    "\n",
+    "Optimization parameters help in optimizing the GPU memory and effectively using the compute resources. Below are few of the parameters that belong to this category. _The optimization parameters differs for each model and are packaged with the model to handle these variations._\n",
+    "1. enable the deepspeed, ORT and LoRA\n",
+    "2. enable mixed precision training\n",
+    "2. enable multi-node training "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Training parameters\n",
+    "training_parameters = dict(\n",
+    "    num_train_epochs=3,\n",
+    "    per_device_train_batch_size=1,\n",
+    "    per_device_eval_batch_size=1,\n",
+    "    learning_rate=2e-5,\n",
+    "    metric_for_best_model=\"exact\",\n",
+    ")\n",
+    "print(f\"The following training parameters are enabled - {training_parameters}\")\n",
+    "\n",
+    "# Optimization parameters - As these parameters are packaged with the model itself, lets retrieve those parameters\n",
+    "if \"model_specific_defaults\" in foundation_model.tags:\n",
+    "    optimization_parameters = ast.literal_eval(\n",
```

Comment:
What if the dictionary is empty?

Reply:
Then no optimizations are enabled. Are you suggesting to enable optimizations even in that case?